### PR TITLE
Environment change

### PR DESCRIPTION
**Pull request by andrewdvt**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,13 +4,96 @@ A new Flutter project.
 
 ## Getting Started
```
Comment (sergembamba) [inline README.md:5]: Let's remove all credentials from this ReadMe.
And let's add information related to the environment switcher

File: ios/Runner.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -494,15 +494,20 @@
...
 				);
 				PRODUCT_BUNDLE_IDENTIFIER = mobile.flutter.pickme;
 				PRODUCT_NAME = "$(TARGET_NAME)";
 				PROVISIONING_PROFILE_SPECIFIER = "";
+				SUPPORTED_PLATFORMS = "iphoneos iphonesimulator";
+				SUPPORTS_MACCATALYST = NO;
+				SUPPORTS_MAC_DESIGNED_FOR_IPHONE_IPAD = NO;
 				SWIFT_OBJC_BRIDGING_HEADER = "Runner/Runner-Bridging-Header.h";
 				SWIFT_VERSION = 5.0;
+				TARGETED_DEVICE_FAMILY = "1,2";
```
Comment (sergembamba) [inline ios/Runner.xcodeproj/project.pbxproj:510]: Family should only be 1, i.e iOS only

File: ios/Runner.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -677,16 +682,21 @@
...
 				PRODUCT_BUNDLE_IDENTIFIER = mobile.flutter.pickme;
 				PRODUCT_NAME = "$(TARGET_NAME)";
 				PROVISIONING_PROFILE_SPECIFIER = "";
+				SUPPORTED_PLATFORMS = "iphoneos iphonesimulator";
+				SUPPORTS_MACCATALYST = NO;
+				SUPPORTS_MAC_DESIGNED_FOR_IPHONE_IPAD = NO;
 				SWIFT_OBJC_BRIDGING_HEADER = "Runner/Runner-Bridging-Header.h";
 				SWIFT_OPTIMIZATION_LEVEL = "-Onone";
 				SWIFT_VERSION = 5.0;
+				TARGETED_DEVICE_FAMILY = "1,2";
```
Comment (sergembamba) [inline ios/Runner.xcodeproj/project.pbxproj:699]: Same as the above

File: ios/Runner.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -704,15 +714,20 @@
...
 				);
 				PRODUCT_BUNDLE_IDENTIFIER = mobile.flutter.pickme;
 				PRODUCT_NAME = "$(TARGET_NAME)";
 				PROVISIONING_PROFILE_SPECIFIER = "";
+				SUPPORTED_PLATFORMS = "iphoneos iphonesimulator";
+				SUPPORTS_MACCATALYST = NO;
+				SUPPORTS_MAC_DESIGNED_FOR_IPHONE_IPAD = NO;
 				SWIFT_OBJC_BRIDGING_HEADER = "Runner/Runner-Bridging-Header.h";
 				SWIFT_VERSION = 5.0;
+				TARGETED_DEVICE_FAMILY = "1,2";
```
Comment (sergembamba) [inline ios/Runner.xcodeproj/project.pbxproj:730]: Same

File: lib/core/PickMeApp.dart [hunk 5]
```diff
@@ -62,8 +66,19 @@ class PickMeApp{
     );
   }
 }
-
-
+saveBaseUrls()async {
```
Comment (sergembamba) [inline lib/core/PickMeApp.dart:69]: can we make private please, _ saveBaseUrls

File: lib/core/PickMeApp.dart [hunk 6]
```diff
@@ -92,8 +107,19 @@ Future<Map<String, dynamic>> readEnvironment(String uri)async{
     return {"error": ex.toString()};
   }
 }
+Future<void> initAPIEnvironment(Map<String,dynamic> data) async{
```
Comment (sergembamba) [inline lib/core/PickMeApp.dart:110]: Please make private all these methods

File: lib/features/add_skills/presentation/add_skills_page.dart [hunk 7]
```diff
@@ -81,19 +82,7 @@ class _AddSkillsPageState extends BasePageState<AddSkillsPage, AddSkillsBloc> {
...
-          }else if(state.profileEntity!.location!.address =="" ){
-            context.router.push(const LocationRoute());
-          }else if(state.profileEntity!.description!.isEmpty){
-            context.router.push(const FinalDetailsRoute());
-          }else if(!state.profileEntity!.subscriptionPaid!) {
-            context.router.push( PaySomeoneWebViewRoute(from: 0));
-          }else{
-            context.router.pushAndPopUntil( BottomNavigationBarRoute(), predicate: (Route<dynamic> route) => false);
-          }
+        routePage(profileEntity: state.profileEntity!, context: context);
```
Comment (sergembamba) [inline lib/features/add_skills/presentation/add_skills_page.dart:85]: I am not a fan of this force unwrap but it's too risky to change this now

File: lib/features/add_skills/presentation/add_skills_page.dart [hunk 8]
```diff
@@ -247,6 +236,22 @@ class _AddSkillsPageState extends BasePageState<AddSkillsPage, AddSkillsBloc> {
     );
   }
 
+  void routePage({required BuildContext context,required ProfileEntity profileEntity }){
```
Comment (sergembamba) [inline lib/features/add_skills/presentation/add_skills_page.dart:239]: make private and remove the extraneous `void`

File: lib/features/apply_for_job/data/repository_impl/apply_for_job_repository_impl/update_job_date_time_repository_impl.dart [hunk 9]
```diff
@@ -0,0 +1,23 @@
+import 'package:dio/dio.dart';
+import 'package:injectable/injectable.dart';
+import 'package:pickme/features/apply_for_job/domain/repository/apply_for_job_repository/update_job_date_time_repository.dart';
+import 'package:pickme/shared/services/remote/api_service/jobs_service/jobs_service.dart';
+
+@Injectable(as: UpdateJobDateTimeRepository)
+class UpdateJobDateTimeRepositoryImpl extends UpdateJobDateTimeRepository {
+      final JobService jobService;
```
Comment (sergembamba) [inline lib/features/apply_for_job/data/repository_impl/apply_for_job_repository_impl/update_job_date_time_repository_impl.dart:8]: Check formatting here

File: lib/features/apply_for_job/data/repository_impl/apply_for_job_repository_impl/update_job_date_time_repository_impl.dart [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+  UpdateJobDateTimeRepositoryImpl({required this.jobService});
+
+  @override
+  Future<bool> call({UpdateJobDateTimeRepositoryParams? params}) async{
+    try{
+      return await jobService.updateJobDateTime(jobEntity: params!.jobEntity, dateAndTime: params!.dateAndTime);
+    } on DioException catch (ex){
+
+      throw(ex.response.toString());
```
Comment (sergembamba) [inline lib/features/apply_for_job/data/repository_impl/apply_for_job_repository_impl/update_job_date_time_repository_impl.dart:17]: Can we add a log entry here

File: lib/features/apply_for_job/domain/entities/date_and_time.dart [hunk 11]
```diff
@@ -0,0 +1,14 @@
+class DateAndTime {
```
Comment (sergembamba) [inline lib/features/apply_for_job/domain/entities/date_and_time.dart:1]: Can we freeze or mark as immutable this model unless it's meant to be mutated

File: lib/features/apply_for_job/domain/repository/apply_for_job_repository/update_job_date_time_repository.dart [hunk 12]
```diff
@@ -0,0 +1,12 @@
+import 'package:pickme/base_classes/base_repository.dart';
+import 'package:pickme/features/apply_for_job/domain/entities/date_and_time.dart';
+import 'package:pickme/shared/domain/entities/job_entity.dart';
+
+abstract class UpdateJobDateTimeRepository  extends BaseRepository<UpdateJobDateTimeRepositoryParams, bool>{}
+
+class UpdateJobDateTimeRepositoryParams extends BaseRepositoryParams{
```
Comment (sergembamba) [inline lib/features/apply_for_job/domain/repository/apply_for_job_repository/update_job_date_time_repository.dart:7]: Check formatting here

File: lib/features/apply_for_job/domain/use_cases/apply_for_job_usecase/update_job_date_time_usecase.dart [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+
+@Injectable()
+class UpdateJobDateTimeUseCase extends BaseUseCase<UpdateJobDateTimeUseCaseParams, bool>{
+
+  final UpdateJobDateTimeRepository updateJobDateTimeRepository;
+
+  UpdateJobDateTimeUseCase({required this.updateJobDateTimeRepository});
+
+  @override
+  Future<bool> call({UpdateJobDateTimeUseCaseParams? params}) async{
```
Comment (sergembamba) [inline lib/features/apply_for_job/domain/use_cases/apply_for_job_usecase/update_job_date_time_usecase.dart:16]: Check formatting and please log errors

File: lib/features/apply_for_job/presentation/apply_for_job_page.dart [hunk 14]
```diff
@@ -49,6 +50,46 @@ class _ApplyForJobPageState extends BasePageState<ApplyForJobPage, ApplyForJobBl
...
+          getBloc().preloaderActive = false;
+          Navigator.pop(context);
+          wErrorPopUp(message: state.error!, type: getLocalization().error, context: context);
+        }
+        if(state is UpdateJobDateTimeState && state.dataState == DataState.success){
+          getBloc().add(RespondToJobInterestEvent(jobEntity: widget.job, status: "booked"));
+        }
+        ///////////////////////////////////////////
+
+        if(state is RespondToJobInterestState && state.dataState == DataState.loading){
```
Comment (sergembamba) [inline lib/features/apply_for_job/presentation/apply_for_job_page.dart:70]: Nothing happening here?

File: lib/features/apply_for_job/presentation/bloc/apply_for_job_bloc.dart [hunk 15]
```diff
@@ -17,10 +20,43 @@ part 'apply_for_job_state.dart';
...
+        on<RespondToJobInterestEvent>((event, emit) => _onRespondToJobInterestEvent(event, emit));
+    }
+
+   Future<void> _onRespondToJobInterestEvent(
+       RespondToJobInterestEvent event,
+        Emitter<ApplyForJobPageState> emit) async{
+
+        emit(RespondToJobInterestState()..dataState=DataState.loading);
+        try{
+
```
Comment (sergembamba) [inline lib/features/apply_for_job/presentation/bloc/apply_for_job_bloc.dart:40]: Extraneous empty line

File: lib/features/apply_for_job/presentation/bloc/apply_for_job_event.dart [hunk 16]
```diff
@@ -18,4 +18,17 @@ class ApplyForJobClickedEvent extends ApplyForJobPageEvent{
...
+class UpdateJobDateTimeEvent extends ApplyForJobPageEvent{
+   final JobEntity jobEntity;
+   final DateAndTime dateAndTime;
+
+   UpdateJobDateTimeEvent({required this.jobEntity, required this.dateAndTime});
+}
+
+class RespondToJobInterestEvent extends ApplyForJobPageEvent{
+   final JobEntity jobEntity;
+  final String status;
```
Comment (sergembamba) [inline lib/features/apply_for_job/presentation/bloc/apply_for_job_event.dart:32]: Check formatting here

File: lib/features/id_reason/presentation/id_reason_page.dart [hunk 17]
```diff
@@ -0,0 +1,213 @@
...
+  const IdReasonPage({super.key});
+
+  @override
+  _IdReasonPageState createState() => _IdReasonPageState();
+}
+
+class _IdReasonPageState extends BasePageState<IdReasonPage, IdReasonBloc> {
+
+  @override
+  void initState() {
```
Comment (sergembamba) [inline lib/features/id_reason/presentation/id_reason_page.dart:27]: Override not needed as there are no added functionality

File: lib/features/id_reason/presentation/id_reason_page.dart [hunk 18]
```diff
@@ -0,0 +1,213 @@
...
+    // TODO: implement initState
+    super.initState();
+
+  }
+
+    @override
+  PreferredSizeWidget? buildAppbar() {
```
Comment (sergembamba) [inline lib/features/id_reason/presentation/id_reason_page.dart:34]: Check formatting

**Review by sergembamba — CHANGES_REQUESTED** [review]

File: lib/core/PickMeApp.dart (hunk 5 above)
Comment (andrewdvt) [reply lib/core/PickMeApp.dart:69]: Done

Comment (sergembamba) [reply lib/core/PickMeApp.dart]: @andrewdvt  not reflecting

File: lib/features/add_skills/presentation/add_skills_page.dart (hunk 7 above)
Comment (andrewdvt) [reply lib/features/add_skills/presentation/add_skills_page.dart:85]: I agree. I'll leave it as is

File: lib/features/apply_for_job/data/repository_impl/apply_for_job_repository_impl/update_job_date_time_repository_impl.dart (hunk 10 above)
Comment (andrewdvt) [reply lib/features/apply_for_job/data/repository_impl/apply_for_job_repository_impl/update_job_date_time_repository_impl.dart:17]: I added a log entry in the exception

File: lib/features/apply_for_job/domain/entities/date_and_time.dart (hunk 11 above)
Comment (andrewdvt) [reply lib/features/apply_for_job/domain/entities/date_and_time.dart:1]: Made the constructor a const constructor

File: lib/features/apply_for_job/domain/repository/apply_for_job_repository/update_job_date_time_repository.dart (hunk 12 above)
Comment (andrewdvt) [reply lib/features/apply_for_job/domain/repository/apply_for_job_repository/update_job_date_time_repository.dart:7]: Not sure what to change here. @sergembamba which part should I update?

File: lib/features/apply_for_job/domain/use_cases/apply_for_job_usecase/update_job_date_time_usecase.dart (hunk 13 above)
Comment (andrewdvt) [reply lib/features/apply_for_job/domain/use_cases/apply_for_job_usecase/update_job_date_time_usecase.dart:16]: Updated

File: lib/features/apply_for_job/presentation/apply_for_job_page.dart (hunk 14 above)
Comment (andrewdvt) [reply lib/features/apply_for_job/presentation/apply_for_job_page.dart:70]: Removed unhandled state

File: lib/features/my_bookings_upcoming/domain/entities/display_entity.dart [hunk 19]
```diff
@@ -0,0 +1,14 @@
+class DisplayEntity{
```
Comment (sergembamba) [inline lib/features/my_bookings_upcoming/domain/entities/display_entity.dart:1]: Are all these properties optional?

File: lib/core/PickMeApp.dart [hunk 20]
```diff
@@ -62,8 +66,19 @@ class PickMeApp{
     );
   }
 }
-
-
+_saveBaseUrls()async {
```
Comment (sergembamba) [inline lib/core/PickMeApp.dart:69]: We can merge this PR, but later create an EnvironmentProvider entity that will encapsulate all related environment responsibilities.
This should follow up on a subsequent PR

**Review by sergembamba — APPROVED** [review]

**Review by sergembamba — APPROVED** [review]

**Pull request merged**